### PR TITLE
Add -e option to pull request

### PR DESCRIPTION
**Pull request by m-nakamura145**
Hi. ```-e``` option improves pull-request with template.

For example 
run ```pull-request -F template_file -e``` then open editor and configure a template file. :tada: 

related issue: https://github.com/github/hub/issues/446

### CODE REVIEW

File: commands/pull_request.go [hunk 1]
```diff
@@ -55,6 +56,7 @@ func init() {
 	cmdPullRequest.Flag.BoolVarP(&flagPullRequestForce, "force", "f", false, "FORCE")
 	cmdPullRequest.Flag.StringVarP(&flagPullRequestFile, "file", "F", "", "FILE")
 	cmdPullRequest.Flag.StringVarP(&flagPullRequestAssignee, "assign", "a", "", "USER")
+	cmdPullRequest.Flag.BoolVarP(&flagPullRequestEditor, "editor", "e", false, "EDITOR")
```
Comment (mislav) [inline commands/pull_request.go]: For parity with `git commit -e`, I think the long name for this flag should be `--edit`, not `--editor`.

File: commands/pull_request.go [hunk 2]
```diff
@@ -186,6 +188,14 @@ func pullRequest(cmd *Command, args *Args) {
 		utils.Check(err)
 	}
 
+	if flagPullRequestEditor {
+		editor, err = github.NewEditor("PULLREQ", "pull request", title + "\n\n" + body)
```
Comment (mislav) [inline commands/pull_request.go]: Why does this repeat the above logic? Couldn't we just have the above conditional branch happen unconditionally if `--edit` was used?

**Comment by mislav** [comment]
Thanks! This will need a test in [`features/pull_request.feature`](https://github.com/github/hub/blob/d45511c7aaca000c49ac1a2ef1a162a6153b0c44/features/pull_request.feature) to get accepted. Look for existing tests using `-F` or "text editor".

File: commands/pull_request.go (hunk 1 above)
Comment (m-nakamura145) [inline commands/pull_request.go]: fd09d23
Fix!! :bow:

File: commands/pull_request.go (hunk 2 above)
Comment (m-nakamura145) [inline commands/pull_request.go]: fd09d23
Oh...sorry, I want to open editor when template file exits. Fix! :bow:

**Comment by m-nakamura145** [comment]
fd09d23
Thanks!! I added test to ``` features/pull_request.feature``` :+1:

**Comment by mislav** [comment]
Thanks for updating! This is getting really close to being done. However, I'm not satisfied with the test completely. We're not verifying in any way that the template was seen by the user and offered for editing. However, setting up a proper test is tricky and I won't blame you if I don't want to deal with that right now. When I catch some time I'll try to work on top of your code to improve the test.

**Comment by michaeljs1990** [comment]
Sorry for jumping in late! this is an awesome addition but to me it seems like this provides a much different feature then -F. It would be preferable to have a completely different flag for this IMO. -F to me is when you have already completely prepared a message where if would like to use a template It would seem like you should be able to do something like -T instead of having to set -F and then -e. It is also just odd to me because you very rarely see command line flags that manipulate another command line flag in a way like this and i think would lead to some confusion among end users.

**Comment by michaeljs1990** [comment]
@m-nakamura145 this is a feature I would very much like are you still working on it? I don't want it to sit in PR purgatory :cry:  so let me know if you are no longer invested in getting this up to current standards and I will take over.

**Comment by m-nakamura145** [comment]
@michaeljs1990 I'm sorry for beging late. :pray:  I will do this PR until the last. :runner:

**Comment by michaeljs1990** [comment]
@m-nakamura145 you have any idea when you will get to make the changes requested above?

**Comment by mislav** [comment]
Do we still want this feature in? :wink:

**Comment by m-nakamura145** [comment]
I want this feature but couldn't write test code. sorry :bow: :weary:

**Comment by mislav** [comment]
@m-nakamura145 That's OK! I'll see what I can do.

**Comment by michaeljs1990** [comment]
I would still very much like this feature :+1:

**Comment by dhilton** [comment]
OK - I'm just looking at the tests - getting 7 failures that are unrelated to this PR, so ignoring for now, assuming they'll go away when rebased against current master.

* To summarise, we need to change the flag from '-f' to '-T'
* The PR needs a working cucumber test to cover the rendering of a commit message given a template

Does it need a golang written test?

**Comment by mislav** [comment]
The most useful tests for this should be added to `feature/pull_request.feature`.

Why changing `-F` to `-T`?

**Comment by mironal** [comment]
:+1:

**Comment by halkeye** [comment]
I see the tests are passing, does it need more tests? Can I help in any way to get this landed?

**Comment by keyan** [comment]
:+1: would really like to see this too, happy to help if help is needed. Looks like a rebase is all it needs though.

**Comment by Tonkpils** [comment]
Is this still valid with https://github.com/github/hub/pull/1187 ?

**Comment by dhilton** [comment]
No I think this has been dealt with thanks!

**Comment by dcosson** [comment]
I'm not sure that this is fully addressed by templates in #1187. With the -F flag, I can use bash process substitution to dynamically generate the PR description based on some script (e.g. as a simple example `hub pull-request -F <(git log --format=%B origin/master..HEAD)`). I wish I could do this as a dynamic default but have the content open in an editor before creating the PR.

Can I accomplish this with the templates?

**Comment by bsander** [comment]
@dcosson probably not, though I've been using a short bash script for a while to accomplish this feature using a tmpfile. You should be able to do it with something like this:

```shell
PR_MSG=$(mktemp --suffix=".md" -t PR_MSG.XXXXXX )
git log --format=%B origin/master..HEAD > "$PR_MSG"
$EDITOR -w "$PR_MSG"
hub pull-request -F "$PR_MSG" 
rm -f "$PR_MSG"
```

**Pull request merged**